runtime/lua/vim/fs.lua

```diff
@@ -308,6 +318,8 @@ function M.find(names, opts)
 
       for other, type_ in M.dir(dir) do
         local f = M.joinpath(dir, other)
+        local stat = vim.uv.fs_stat(f)
```

Comment:
this is a perf cost, which should only be paid if it's actually necessary (i.e. depending on the `follow` param)

Reply:
Ok, I'll change it